frontend/apps/core-app/src/components/Web/Input.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+  This input is designed to be used as a date, month, or week input on web only.
+  It copies some code from native base so that we can apply the input style to a html input tag.
+*/
+
+import * as React from 'react';
+import {
+  Box,
+  ITheme,
+  usePropsResolution,
+  useStyledSystemPropsResolver,
+  useTheme,
+  useThemeProps,
+  useToken,
+} from 'native-base';
+import { useHover } from '@react-native-aria/interactions';
+import styled from 'styled-components';
+
+type Props = {
+  type: string;
+  value: string;
+  invalid: boolean;
+  onChange: (value: string) => void;
+};
+
+// native-base doesn't export this function
+const makeStyledComponent = (Comp: any) => {
+  // eslint-disable-next-line react/display-name
+  return React.forwardRef((props: any, ref: any) => {
+    const [style, restProps] = useStyledSystemPropsResolver(props);
+    return (
+      <Comp {...restProps} style={style} ref={ref}>
+        {props.children}
+      </Comp>
+    );
+  });
+};
+
+// native-base doesn't export this function
+function useResolvedFontFamily(props: {
+  fontFamily?: keyof ITheme['fonts'];
+  fontStyle?: string;
+  fontWeight?: keyof ITheme['fontWeights'];
+}) {
+  const { fontFamily, fontStyle, fontWeight } = props;
+  let newFontFamily = fontFamily;
+  let newFontStyle = fontStyle;
+  let newFontWeight = fontWeight;
+
+  const { fontConfig, fontWeights, fonts } = useTheme();
+  if (fontWeight && fontStyle && fontFamily && fontFamily in fonts) {
+    // TODO: Fix typing remove any.
+    const fontToken: any = fonts[fontFamily];
+    if (fontConfig && fontConfig[fontToken]) {
+      // If a custom font family is resolved, set fontWeight and fontStyle to undefined.
+      // https://github.com/GeekyAnts/NativeBase/issues/3811
+      // On Android, If a fontFamily and fontWeight both are passed, it behaves in a weird way and applies system fonts with passed fontWeight. This happens only for some fontWeights e.g. '700' or 'bold'. So, if we find a custom fontFamily, we remove fontWeight and fontStyle
+      // @ts-ignore
+      newFontWeight = undefined;
+      // @ts-ignore
+      newFontStyle = undefined;
+
+      const fontWeightNumber =
+        fontWeight in fontWeights ? fontWeights[fontWeight] : fontWeight;
+      const fontVariant = fontConfig[fontToken][fontWeightNumber];
+
+      if (typeof fontVariant === 'object') {
+        if (fontVariant[fontStyle]) newFontFamily = fontVariant[fontStyle];
+      } else {
+        newFontFamily = fontVariant;
+      }
+    } else {
+      newFontFamily = fonts[fontFamily];
+    }
+  }
+
+  return {
+    fontFamily: newFontFamily,
+    fontWeight: newFontWeight,
+    fontStyle: newFontStyle,
+  };
+}
+
+const StyledWrapper = makeStyledComponent(Box);
+
+export const Input: React.FC<Props> = ({
+  type: t,
+  value,
+  invalid,
+  onChange,
+  ...otherProps
+}) => {
+  const StyledInput = styled.input`
```

Comment:
don't these typically live outside of the components?

Reply:
Yeah this was to prevent the import breaking on mobile, but I've fixed this in some local commits